homeassistant/components/esphome/media_player.py

```diff
@@ -0,0 +1,154 @@
+"""Support for ESPHome media players."""
+from __future__ import annotations
+
+from aioesphomeapi import (
+    MediaPlayerCommand,
+    MediaPlayerEntityState,
+    MediaPlayerInfo,
+    MediaPlayerState,
+)
+
+from homeassistant.components import media_source
+from homeassistant.components.media_player import (
+    MediaPlayerDeviceClass,
+    MediaPlayerEntity,
+)
+from homeassistant.components.media_player.browse_media import (
+    BrowseMedia,
+    async_process_play_media_url,
+)
+from homeassistant.components.media_player.const import MediaPlayerEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_IDLE, STATE_PAUSED, STATE_PLAYING
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import EsphomeEntity, EsphomeEnumMapper, platform_async_setup_entry
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up esphome media players based on a config entry."""
+    await platform_async_setup_entry(
+        hass,
+        entry,
+        async_add_entities,
+        component_key="media_player",
+        info_type=MediaPlayerInfo,
+        entity_type=EsphomeMediaPlayer,
+        state_type=MediaPlayerEntityState,
+    )
+
+
+_STATES: EsphomeEnumMapper[MediaPlayerState, str] = EsphomeEnumMapper(
+    {
+        MediaPlayerState.IDLE: STATE_IDLE,
+        MediaPlayerState.PLAYING: STATE_PLAYING,
+        MediaPlayerState.PAUSED: STATE_PAUSED,
+    }
+)
+
+
+class EsphomeMediaPlayer(
+    EsphomeEntity[MediaPlayerInfo, MediaPlayerEntityState], MediaPlayerEntity
+):
+    """A media player implementation for esphome."""
+
+    # _attr_should_poll = False
+    _attr_device_class = MediaPlayerDeviceClass.SPEAKER
+
+    @property
+    def state(self) -> str | None:
+        """Return current state."""
+        return _STATES.from_esphome(self._state.state)
+
+    @property
+    def is_volume_muted(self) -> bool:
+        """Return true if volume is muted."""
+        if not self._static_info.supports_mute:
```

Comment:
why wouldn't we always support mute? It would just be "playing into the void" / 0 volume?

Reply:
Just in case there is some hardware with no volume/mute control at all...

Reply:
I guess we can assume all audio output have some way to mute or at least control volume. (Would not be very useful if it didn't IMO)